### PR TITLE
Feat/nomad odr namespace fix

### PR DESCRIPTION
**Pull request by paladin-devops**
Addresses #2791.

The Nomad API client's namespace configuration will be set equal to the namespace in the jobspec, for Nomad jobspec deployments. The image below showcases the desired result of a jobspec deployment with `namespace = "dev"` set in its jobspec, where the Waypoint server & static runner are deployed to a namespace (in this case, default), the ODR is deployed to a namespace (in this case, default), and the actual application (in this case, nomad-test) is deployed to a separate namespace, dev, by the ODR.

image

My ODR profile, for reference:
```json
{
	"datacenter": "dc1",
	"namespace": "default",
	"nomad_host": "http://my-nomad-host:4646",
	"region": "global",
	"resources_cpu": "200",
	"resources_memory": "600"
}
```
Excerpt from my jobspec, for reference:
```hcl
job "nomad-test" {
  type = "service"
  namespace = "dev"
```

### CODE REVIEW

**Review by briancain — COMMENTED** [review]
This change looks good to me, pretty easy fix! If you could, please rebase the commit history down to just the change (i.e. remove the merge commits) and add a `bug` fix [changelog .txt file](https://github.com/hashicorp/waypoint/blob/main/.github/CHANGELOG_GUIDE.md)! Thank you!

File: builtin/nomad/jobspec/platform.go [hunk 1]
```diff
@@ -83,6 +83,9 @@ func (p *Platform) Deploy(
 	// Set our deployment ID on the meta.
 	job.SetMeta(metaId, result.Id)
 
+	// Update our client to use the Namespace set in the jobspec
+	client.SetNamespace(*job.Namespace)
```
Comment (briancain) [inline builtin/nomad/jobspec/platform.go:87]: Does this need to be a pointer here, or can we use the `job.Namepsace` string value directly? 🤔

**Review by briancain — APPROVED** [review]
Looks good to me! Just had a minor question. Thanks for taking the time to open this pull request and fixup the issue you reported! 🙏🏻

File: builtin/nomad/jobspec/platform.go (hunk 1 above)
Comment (evanphx) [reply builtin/nomad/jobspec/platform.go:87]: We checked and it is a pointer already, @briancain was mistaken.

**Pull request merged**